catroid/src/androidTest/java/org/catrobat/catroid/test/web/ServerCallsTest.java

```diff
@@ -101,22 +100,20 @@ public void testRegisterWithExistingUser() {
 					"de", "at", token, getInstrumentation().getTargetContext());
 
 			Log.i(TAG, "user registered: " + userRegistered);
-			assertTrue("Should be a new user, but server response indicates that this user already exists",
-					userRegistered);
+			assertTrue(userRegistered);
 
 			SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getInstrumentation().getTargetContext());
 			token = sharedPreferences.getString(Constants.TOKEN, Constants.NO_TOKEN);
 			ServerCalls.getInstance().register(testUser, testPassword, testEmail, "de",
 					"at", token, getInstrumentation().getTargetContext());
 
-			assertFalse("should never be reached because the user is already registered and can't be registered again",
-					true);
+			assertFalse(true);
 		} catch (WebconnectionException e) {
 			Log.i(TAG, "Webconnection error (this is expected behaviour)", e);
-			assertTrue("an exception should be thrown because the user(-e-mail) is already registered", true);
-			assertEquals("wrong status code from server", STATUS_CODE_USER_ADD_EMAIL_EXISTS, e.getStatusCode());
-			assertNotNull("no error message available", e.getMessage());
-			assertTrue("no error message available", e.getMessage().length() > 0);
+			assertTrue(true);
```

Comment:
Even better than `assertFalse(true);` :rofl:

Reply:
btw, PMD has a rule for asserts like that, [UnnecessaryBooleanAssertion](https://pmd.github.io/pmd-5.8.1/pmd-java/rules/java/junit.html#UnnecessaryBooleanAssertion) (same in [latest version](https://pmd.github.io/pmd-6.4.0/pmd_rules_java_errorprone.html#unnecessarybooleanassertion)).
It doesn't work with `AndroidTestCase` :crying_cat_face:, however, once all tests are ported to JUnit4, this and all other awesome pmd junit rules will work again. :smile_cat:

Reply:
pmd would have rules for pretty much all my problems here (assertTrue used for checking equality, or against null, or true for that matter), assertion messages. but.... most of those only come from pmd4+ ruleset, while were still using 2.something.

im slowly working my way through the unit tests, next is going to be cleanup of said assertions (and the ones mentioned above), and then porting all tests to junit4....

Reply:
We're using `pmd version="5.8.1"` btw :wink:

Reply:
sorry for the ambiguity, i was referring to the pmd ruleset...

Reply:
...or maybe im messing up here, anyways, i was searching for pmd rules to find all bad assertions and ran against a wall  because of some version issue...